### ISSUE TITLE
UTF in tool output breaks sqlite backend

### ISSUE THREAD

**Issue by mr-c** [galaxyproject/galaxy#714]
```
ProgrammingError: (pysqlite2.dbapi2.ProgrammingError) You must not use 8-bit bytestrings unless you use a text_factory that can interpret 8-bit bytestrings (like text_factory = str). It is highly recommended that you instead just switch your application to Unicode strings. [SQL: u
'UPDATE history_dataset_association SET update_time=?, info=?, blurb=?, peek=?, tool_version=? WHERE history_dataset_association.id = ?'] [parameters: ('2015-09-10 21:12:57.166823', u'\nFatal error: Exit code 2 ()\n\n|| This is the script abundance-dist-single.py in khmer.\n|| You
 are running khmer version 2.0\n|| You are also using screed version 0.9\n||\n|| If you use this script in a publication, please cite EACH of the following:\n||\n||   ', 'error', 'no peek', '\n|| This is the script abundance-dist-single.py in khmer.\n|| You are running khmer versi
on 2.0\n|| You are also using screed version 0.9\n||\n|| If you use this script in a publication, please cite EACH of the following:\n||\n||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\n||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0
101271\n||   * A. D\xc3\xb6ring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\n||\n|| Please see http://khmer.readthedocs.org/en/latest/citations.html for details.\n\nkhmer 2.0\n', 10)]
```

**Comment by dannon** [comment]
Thanks for the report, I'll take a look since I'm working on another utf8 error (though this one is related to mysql) right now.

**Comment by mr-c** [comment]
Coo'. The source is at https://github.com/mr-c/tools-iuc/tree/khmer-wrappers if you want to reproduce

**Comment by dannon** [comment]
Even better, thanks!

**Comment by mr-c** [comment]
Here's another unicode failure, this time with running a test using `planemo test`/`run_tests.sh`

```
======================================================================
ERROR: Abundance Distribution (all-in-one) ( gedlab-khmer-abundance-dist-single ) > Test-2
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/galaxy/eggs/nose-1.3.6-py2.7.egg/nose/case.py", line 133, in run
    self.runTest(result)
  File "/home/ubuntu/galaxy/eggs/nose-1.3.6-py2.7.egg/nose/case.py", line 151, in runTest
    test(result)
  File "/usr/lib/python2.7/unittest/case.py", line 395, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib/python2.7/unittest/case.py", line 335, in run
    result.addFailure(self, sys.exc_info())
  File "/home/ubuntu/galaxy/eggs/nose-1.3.6-py2.7.egg/nose/proxy.py", line 146, in addFailure
    plugins.addFailure(self.test, err)
  File "/home/ubuntu/galaxy/eggs/nose-1.3.6-py2.7.egg/nose/plugins/manager.py", line 99, in __call__
    return self.call(*arg, **kw)
  File "/home/ubuntu/galaxy/eggs/nose-1.3.6-py2.7.egg/nose/plugins/manager.py", line 167, in simple
    result = meth(*arg, **kw)
  File "/home/ubuntu/galaxy/eggs/NoseHTML-0.4.1-py2.7.egg/nosehtml/plugin.py", line 93, in addFailure
    self.print_test( 'failure', test, '\n'.join( traceback.format_exception( *error ) ) )
  File "/home/ubuntu/galaxy/eggs/NoseHTML-0.4.1-py2.7.egg/nosehtml/plugin.py", line 67, in print_test
    print >> f.file, "<div id='capture_%d' style='display: none'><pre class='capture'>%s</pre></div>" % ( f.counter, cgi.escape( test.capturedOutput ) )
UnicodeEncodeError: 'ascii' codec can't encode character u'\xf6' in position 1301: ordinal not in range(128)
```

**Comment by jmchilton** [comment]
Attempted to fix this in this release cycle but it seems like maybe it still isn't fixed, I'm going to set a milestone of 16.04 for this and hopefully revisit my attempts to fix this again this release cycle.

**Comment by martenson** [comment]
please provide reliable testcase

**Comment by mr-c** [comment]
@martenson non-ASCII in a tool output did it for me.

**Comment by martenson** [comment]
@mr-c I am re-openning this and adding to 16.07 milestone
thanks

**Comment by martenson** [comment]
this is a bug but the fix did not make it to 16.07 (yet) so I am pushing the issue to 16.10 as this is not a blocking issue

**Comment by jmchilton** [comment]
Okay - so I don't think this is a problem anymore. I checked out the latest version of those tools - which now pass. I changed the exit code detection so they would run and fail. I added some unicode to the standard output since I didn't see anything obvious. Planemo ran fine, Galaxy ran fine, and a report was [produced](http://galaxyproject.github.io/planemo/tool_test_viewer.html?test_data_url=https://gist.githubusercontent.com/jmchilton/9f432a4d82982dac8c6210e71771c9c8/raw/b5499006da0ac118a7b77bb951620488851c7ea2/index). There have been minor updates to the nose lib versions listed in the stack trace and perhaps the CGI escape has been updated also. It seems to all work for me now though.

Feel free to re-open if you can consistently reproduce against a recent release of Galaxy and thanks for pushing on this issue. I think we are getting better.
